src/ban.cpp

```diff
@@ -83,22 +84,22 @@ bool IOBan::isAccountBanned(uint32_t accountId, BanInfo& banInfo)
 	return true;
 }
 
-bool IOBan::isIpBanned(uint32_t clientIP, BanInfo& banInfo)
+bool IOBan::isIpBanned(Connection::Address clientIP, BanInfo& banInfo)
 {
-	if (clientIP == 0) {
+	if (clientIP.is_unspecified()) {
 		return false;
 	}
 
 	Database& db = Database::getInstance();
 
-	DBResult_ptr result = db.storeQuery(fmt::format("SELECT `reason`, `expires_at`, (SELECT `name` FROM `players` WHERE `id` = `banned_by`) AS `name` FROM `ip_bans` WHERE `ip` = {:d}", clientIP));
+	DBResult_ptr result = db.storeQuery(fmt::format("SELECT `reason`, `expires_at`, (SELECT `name` FROM `players` WHERE `id` = `banned_by`) AS `name` FROM `ip_bans` WHERE `ip` = INET6_ATON({:d})", db.escapeString(clientIP.to_string())));
```

Comment:
```suggestion
	DBResult_ptr result = db.storeQuery(fmt::format("SELECT `reason`, `expires_at`, (SELECT `name` FROM `players` WHERE `id` = `banned_by`) AS `name` FROM `ip_bans` WHERE `ip` = INET6_ATON({})", db.escapeString(clientIP.to_string())));
```
this probably isn't going to be int anymore?